lib/adal-angular.js

```diff
@@ -320,7 +320,7 @@ if (typeof module !== 'undefined' && module.exports) {
                             if (authService.loginInProgress()) {
                                 authService.info('login already start.');
                                 return $q.reject();
-                            } else if (authService.config && isEndpoint) {
+                            } else if (authService.config && (!authService.config.endpoints || isEndpoint)) {
```

Comment:
@tushargupta51 I don't see the client id matching.. Noted below the lines I'm looking.  Can you please explain, how the first one is including the 'client id' check?

In adal.js, line#769-776

AuthenticationContext.prototype.getResourceForEndpoint = function (endpoint) {
    if (this.config && this.config.endpoints) {
        for (var configEndpoint in this.config.endpoints) {
            // configEndpoint is like /api/Todo requested endpoint can be /api/Todo/1
            if (endpoint.indexOf(configEndpoint) > -1) {
                return this.config.endpoints[configEndpoint];
            }
        }
    }

in adal-angular.js, line#311-317

if (authService.config) {
   for (var endpointUrl in authService.config.endpoints) {
   if (config.url.indexOf(endpointUrl) > -1) {
      isEndpoint = true;
   }
 }
}